osf/management/commands/sync_doi_metadata.py

```diff
@@ -1,41 +1,77 @@
 #!/usr/bin/env python3
+import time
 import datetime
 import logging
 
 from django.core.management.base import BaseCommand
+from datacite.errors import DataCiteServerError
+from website.identifiers.clients.exceptions import CrossRefRateLimitError
 from osf.models import Identifier
 
 from framework.celery_tasks import app
 
 logger = logging.getLogger(__name__)
 
 
-@app.task(name='osf.management.commands.sync_doi_metadata', max_retries=5, default_retry_delay=60)
-def sync_identifier_doi(identifier_id):
+RATE_LIMIT_RETRY_DELAY = 60 * 5
+
+
+@app.task(name='osf.management.commands.sync_doi_metadata', max_retries=5, default_retry_delay=RATE_LIMIT_RETRY_DELAY)
+def sync_identifier_doi(identifier_id, delay=0):
+    time.sleep(delay)
+
     identifier = Identifier.objects.get(id=identifier_id)
-    identifier.referent.request_identifier_update('doi')
-    identifier.save()
-    logger.info(f' doi update for {identifier.value} complete')
+    try:
+        identifier.referent.request_identifier_update('doi')
+        identifier.save()
+        logger.info(f' doi update for {identifier.value} complete')
+    except CrossRefRateLimitError as err:
+        logger.warning(f'Doi update for {identifier.value} failed because of rate limit: {err}')
+        raise
+    except DataCiteServerError as err:
+        # the first param is status code, the second one is text
+        # see create_identifier.metadata_post call in datacite.py
+        status_code, text = err.args
+        if status_code == 429:
+            logger.warning(f'Doi update for {identifier.value} failed because of rate limit: {text}')
+        else:
+            logger.warning(f'Doi update for {identifier.value} failed. Error: {text}')
 
+        raise
+    except Exception as err:
+        logger.warning(f'Doi update for {identifier.value} failed because of an unexpected error: {err}')
+        raise
```

Comment:
Instead of failing, can we wait the amount of time they specify (there may be something returned from the API to indicate the time to wait, but otherwise just the time they say in the API docs) and then try again and continue the script?